src/snowcli/snow_connector.py

```diff
@@ -35,15 +38,33 @@ def connect_to_snowflake(temporary_connection: bool = False, connection_name: Op
             {k: v for k, v in overrides.items() if v is not None}
         )
 
+    private_key = None
+    if "private_key_path" in connection_parameters:
+        if connection_parameters.get("authenticator") == "SNOWFLAKE_JWT":
+            private_key_passphrase = os.getenv("PRIVATE_KEY_PASSPHRASE", None)
+            private_key = load_pem_to_der(connection_parameters["private_key_path"])
+            del connection_parameters["private_key_path"]
+        else:
+            raise Exception(
```

Comment:
```suggestion
            raise ClickException(
```
We use `ClickException` to have nice handling and formatting.